valhalla/loki/reach.h

```diff
@@ -75,7 +75,7 @@ class Reach : public thor::Dijkstras {
   void enqueue(const baldr::GraphId& node_id,
                baldr::GraphReader& reader,
                const std::shared_ptr<sif::DynamicCost>& costing,
-               const baldr::GraphTile*& tile);
+               const baldr::GraphTile* tile);
```

Comment:
this is the fix, we dont let enqueue modify the pointer outside of its scope. if we do the filter function can get the wrong tile in the next loop iteration: https://github.com/valhalla/valhalla/pull/2597/files#diff-71585ca21c04f5620b9c09883c71b910R65